fs/layer/node.go

```diff
@@ -62,26 +64,32 @@ func newNode(layerDgst digest.Digest, r reader.Reader, blob remote.Blob) (fusefs
 		return nil, fmt.Errorf("failed to get a TOCEntry of the root")
 	}
 	return &node{
-		r: r,
-		e: root,
-		s: newState(layerDgst, blob),
+		r:        r,
+		e:        root,
+		s:        newState(layerDgst, blob),
+		layerSha: layerDgst,
 	}, nil
 }
 
 // node is a filesystem inode abstraction.
 type node struct {
 	fusefs.Inode
-	r      reader.Reader
-	e      *estargz.TOCEntry
-	s      *state
-	opaque bool // true if this node is an overlayfs opaque directory
+	r        reader.Reader
+	e        *estargz.TOCEntry
+	s        *state
+	layerSha digest.Digest
+	opaque   bool // true if this node is an overlayfs opaque directory
 }
 
 var _ = (fusefs.InodeEmbedder)((*node)(nil))
 
 var _ = (fusefs.NodeReaddirer)((*node)(nil))
 
 func (n *node) Readdir(ctx context.Context) (fusefs.DirStream, syscall.Errno) {
+	// Measure how long node_readdir operation takes.
+	start := time.Now() // set start time
+	defer commonmetrics.MeasureLatency(commonmetrics.NodeReaddir, n.layerSha, start)
+
```

Comment:
Maybe we should measure this also in nanosecond scale because most of readdir operation will take < 1ms. 
WDYT?

(not need to be addressed in this PR. can be done in following up PRs)

Reply:
I think it does make sense. For all operations, which are sub-millisecond we can look into using either microseconds or nanoseconds. 
As the part of the following PRs I plan to measure which scale works best for that and will get back with the approach which fits best based on my findings.